packages/components/src/components/wallet-icon/wallet-icon.tsx

```diff
@@ -1,34 +1,80 @@
-import React, { CSSProperties } from 'react';
-import { GradientBackgroundTwoPoint } from '../gradient-background-two-point';
+import React from 'react';
+import classNames from 'classnames';
+import { isCryptocurrency } from '@deriv/shared';
+import Icon from '../icon';
 import './wallet-icon.scss';
 
-type TWalletIcon = {
-    icon: JSX.Element;
-    primaryColor: CSSProperties['backgroundColor'];
-    secondaryColor: CSSProperties['backgroundColor'];
-    size?: 'small' | 'medium' | 'large';
+type TWalletIconProps = {
+    currency: string;
+    has_bg?: boolean;
+    icon: string;
+    size: string;
+    type: string;
 };
 
-const getBlurRadius = (size: string) => {
-    if (size === 'small') return 16;
-    if (size === 'medium') return 24;
-
-    return 48;
+const sizes: any = {
+    fiat: {
+        xsmall: 12,
+        small: 16,
+        medium: 24,
+        large: 32,
+        xlarge: 48,
+    },
+    crypto: {
+        xxsmall: {
+            width: 26,
+            height: 16,
+        },
+        xsmall: {
+            width: 32,
+            height: 20,
+        },
+        small: {
+            width: 40,
+            height: 24,
+        },
+        medium: {
+            width: 48,
+            height: 30,
+        },
+        large: {
+            width: 64,
+            height: 40,
+        },
+        xlarge: {
+            width: 80,
+            height: 48,
+        },
+        xxlarge: {
+            width: 96,
+            height: 60,
+        },
+    },
 };
 
-const WalletIcon = ({ icon, primaryColor, secondaryColor, size = 'medium' }: TWalletIcon) => {
+const WalletIcon = ({ currency, has_bg = false, icon, size = 'medium', type = 'fiat' }: TWalletIconProps) => {
+    const is_fiat = !isCryptocurrency(currency || '') && currency !== 'USDT';
```

Comment:
@hamid-deriv just for my understanding, why not use props.type to check is_fiat, since we need to pass fiat or crypto to the WalletIcon? Smth like this:
```suggestion
    const is_fiat = type === 'fiat';
```

Reply:
That's a good idea. Refactored to use `type` instead of `is_fiat` 👍